src/mongo/delegates/Views.class.php

```diff
@@ -599,8 +600,37 @@ protected function doJoins($source, $joins, &$dest, $from, $contextAlias, $build
                     {
                         // make sure any sequences are expanded before extracting properties
                         if (isset($ruleset['joins'])) $this->expandSequence($ruleset['joins'],$linkMatch);
-
-                        $dest[_GRAPHS][] = $this->extractProperties($linkMatch,$ruleset,$from);
+                        if (isset($ruleset['filter']))
+                        {
+                            foreach($ruleset['filter'] as $filterPredicate => $filter)
+                            {
+                                foreach($filter as $filterType => $filterMatch)
+                                {
+                                    if (isset($linkMatch[$filterPredicate]))
```

Comment:
What I meant by my comment was that you were right and I was wrong.  We _can_ have a mix of aliases and fquris in the database, but it's not going to make any difference here, because we'd only be matching whatever is specified in the spec.